ledger-service/http-json/src/it/scala/http/HttpServiceIntegrationTest.scala

```diff
@@ -66,6 +78,149 @@ abstract class HttpServiceIntegrationTest
           }
         }: Future[Assertion]
   }
+
+  private val iiouIfaceID: domain.TemplateId.OptionalPkg = domain.TemplateId(None, "IIou", "IIou")
+  "pick up new package's inherited interfaces" in withHttpService { (uri, encoder, _, _) =>
+    import json.JsonProtocol._
+    def createIouAndExerciseTransfer(
+        initialTplId: domain.TemplateId.OptionalPkg,
+        exerciseBy: domain.TemplateId.OptionalPkg,
+    ) = for {
+      aliceH <- getUniquePartyAndAuthHeaders(uri)("Alice")
+      (alice, aliceHeaders) = aliceH
+      createTest <- postCreateCommand(
+        iouCommand(alice, initialTplId),
+        encoder,
+        uri,
+        aliceHeaders,
+      )
+      testIIouID = {
+        discard { createTest._1 should ===(StatusCodes.OK) }
+        createTest._2.convertTo[domain.OkResponse[domain.ActiveContract[JsValue]]].result.contractId
+      }
+      bobH <- getUniquePartyAndAuthHeaders(uri)("Bob")
+      (bob, _) = bobH
+      exerciseTest <- postJsonRequest(
+        uri withPath Uri.Path("/v1/exercise"),
+        encodeExercise(encoder)(
+          iouTransfer(domain.EnrichedContractId(Some(exerciseBy), testIIouID), bob)
+        ),
+        aliceHeaders,
+      )
+    } yield exerciseTest._1 should ===(StatusCodes.OK)
+
+    for {
+      _ <- uploadPackage(uri)(ciouDar)
+      // first, use IIou only
+      _ <- createIouAndExerciseTransfer(
+        initialTplId = domain.TemplateId(None, "IIou", "TestIIou"),
+        // whether we can exercise by interface-ID
+        exerciseBy = iiouIfaceID,
```

Comment:
ok I see already how things blow up regarding naming here `InterfaceIouIfaceID`. This is frankly an unfortunate situation :facepalm: However this can be fixed later, I need to sit down first and think about something simple yet better for naming this interface stuff.

Reply:
I feel like I've hooked your brain too much merely by bothering to include the letters `C` and `I`, and we would be better off had I called these type declarations `Alpha`, `Beta`, `Gamma`, and of course the choice `Delta`.

I have one note on any naming thoughts you have: if an approach is akin to putting the word `Contract` or `Template` in the name of your template definitions, things may have degenerated at some point.